libs/api/domains/syslumenn/src/lib/syslumenn.service.ts

```diff
@@ -1,47 +1,135 @@
-import { SyslumennClient } from './client/syslumenn.client'
 import { Homestay, mapHomestay } from './models/homestay'
 import {
   SyslumennAuction,
   mapSyslumennAuction,
 } from './models/syslumennAuction'
-import { Injectable } from '@nestjs/common'
-import { Person, Attachment, DataUploadResponse } from './models/dataUpload'
+import { Injectable, Inject } from '@nestjs/common'
+import {
+  DataUploadResponse,
+  constructUploadDataObject,
+} from './models/dataUpload'
+
+import { Attachment, Person } from './dto/uploadData.input'
 import {
   OperatingLicense,
   mapOperatingLicense,
 } from './models/operatingLicense'
+import {
+  SyslumennApi,
+  SyslumennApiConfig,
+  VirkarHeimagistingar,
+  Uppbod,
+  VirkLeyfi,
+  Vottord,
+} from '@island.is/clients/syslumenn'
+import { Auth, AuthMiddleware } from '@island.is/auth-nest-tools'
+import {
+  CertificateInfoRepsonse,
+  CertificateRepsonse,
+  mapCertificateInfo,
+} from './models/certificateInfo'
 
+const SYSLUMENN_CLIENT_CONFIG = 'SYSLUMENN_CLIENT_CONFIG'
 @Injectable()
 export class SyslumennService {
-  constructor(private syslumennClient: SyslumennClient) {}
+  private id = ''
+  private accessToken = ''
+
+  constructor(
+    private syslumennApi: SyslumennApi,
+    @Inject(SYSLUMENN_CLIENT_CONFIG)
+    private clientConfig: SyslumennApiConfig,
+  ) {}
+
+  private async login() {
+    const config = {
+      notandi: this.clientConfig.username,
+      lykilord: this.clientConfig.password,
+    }
+
+    const { audkenni, accessToken } = await this.syslumennApi.innskraningPost({
+      notandi: config,
+    })
+    if (audkenni && accessToken) {
+      this.id = audkenni
+      this.accessToken = accessToken
+    }
+  }
+  private async syslumennApiWithAuth() {
+    await this.login()
+
+    const auth = {
```

Comment:
can this by typed, instead of casting the value?

Reply:
I don't know why this doesn't  but the lates commits address this